src/ipc/ipc4/handler.c

```diff
@@ -843,11 +843,13 @@ static int ipc4_module_process_d0ix(struct ipc4_message_request *ipc4)
 		return IPC4_INVALID_RESOURCE_ID;
 	}
 
-	/* only module 0 can be used to set d0ix state */
+	/* only module 0 can be used to set d0ix state.
+	 * TODO: add secondary core support
+	 */
 	if (d0ix.extension.r.prevent_power_gating)
-		pm_runtime_get(PM_RUNTIME_DSP, PLATFORM_PRIMARY_CORE_ID);
+		pm_runtime_disable(PM_RUNTIME_DSP, PLATFORM_PRIMARY_CORE_ID);
 	else
-		pm_runtime_put(PM_RUNTIME_DSP, PLATFORM_PRIMARY_CORE_ID);
+		pm_runtime_enable(PM_RUNTIME_DSP, PLATFORM_PRIMARY_CORE_ID);
```

Comment:
This is potentially good, but I'll need a bit more time to review this. Is this @tmleman  good for you and does not interfere with your PM enablement? We should really get away from calling the SOF pm_runtime_ interfaces, but this needs to be done incrementally.